File: iotOnGolang/internal/model/DeviceVendor.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// DeviceVendor Declaration
//==============================================================
type DeviceVendor struct {
    gorm.Model
     Name                                    string
    LegalName                                    string
    HeadquartersCountry                                    string
    Website                                    string
     DeviceModels           []DeviceModel `gorm:"foreignKey:DeviceModelsFromDeviceVendorId"`
     FirmwareReleases           []FirmwareRelease `gorm:"foreignKey:FirmwareReleasesFromDeviceVendorId"`
     HardwareModules           []HardwareModule `gorm:"foreignKey:HardwareModulesFromDeviceVendorId"`

// parent associations as their child

}

File: iotonrails/app/models/devicevendor.rb

class DeviceVendor < ApplicationRecord


  has_many :DeviceModels, class_name: 'DeviceModel'
  has_many :FirmwareReleases, class_name: 'FirmwareRelease'
  has_many :HardwareModules, class_name: 'HardwareModule'

end
